### DISCUSSION TITLE
esp32-c3 wake up on pin and pin notification - how to contribute back?

### DISCUSSION THREAD

**Discussion by m-cas** [micropython/micropython#15479] [ESP32]
Hi
for a project of mine I fixed the dep sleep  pin wakeup (using gpio pins) for the esp32-c3 platform and also added the capability to return the source pin of the wakeup. I took inspiration and some of the code from PR 7990 and PR 13333 and merged all using the latest master code as per last friday (the two PRs were based on rather old code).
I know how to program and create my own version of micropython but I never worked with git/github (beyond a "git clone" of the repo) so I am incapable of creating a branch and a pull request (if that is the right way at all) to give back the 4 files I did change for integration into the main codeline.
If someone would like to share some instructions / pointers on how to do that in this repo I would be happy to contribute back.

**Comment by Rumidom** [comment]
Normally you create and push to a new branch on the repository (if you own the repository) then you do the rest on the github website itself like its explained in this video: https://www.youtube.com/watch?v=jRLGobWwA3Y , I never did it myself but here I think you should first fork the repository and push to it then you should be able to choose the fork on the pull request form, there are some guidelines specific to micropython that must be followed in the docs: https://github.com/micropython/micropython/wiki/ContributorGuidelines

**Reply by ricksorensen** [reply]
@m-cas I too have difficuties with git and the change process.   Here is a link to some notes I made after the last pull request I did that _seemed_ to get me through.  Maybe it will help.
https://github.com/ricksorensen/micropython-fun/blob/main/gitnotes.md
